docs/core-concepts/message-structure.mdx

```diff
@@ -14,12 +14,13 @@ Each message must specify a `role` that identifies the sender. Valid role format
 
 ## Message Parts
 
-| Attribute                  | Required | Description                                                               |
-| :------------------------- | :------- | :------------------------------------------------------------------------ |
-| `content_type`             | Yes      | MIME type (e.g., `text/plain`, `image/png`)                               |
-| `content` OR `content_url` | Yes      | Content `inline` or via `URL`                                             |
-| `content_encoding`         | No       | `"plain"`  (default) or `"base64"`                                        |
-| `name`                     | No       | Makes this part an **Artifact**                                               |
+| Attribute                  | Required | Description                                                  |
+| :------------------------- | :------- | :----------------------------------------------------------- |
+| `content_type`             | Yes      | MIME type (e.g., `text/plain`, `image/png`)                  |
+| `content` OR `content_url` | Yes      | Content `inline` or via `URL`                                |
+| `content_encoding`         | No       | `"plain"`  (default) or `"base64"`                           |
+| `name`                     | No       | Makes this part an **Artifact**                              |
+| `metadata`                 | No       | Additional metadata like citations or trajectory information |
```

Comment:
```suggestion
| `metadata`                 | No       | Additional metadata to provide additional context or semantic information. |
```

just to avoid maintenence.